### PR TITLE
Update migrator implementation

### LINKED ISSUE #117: Update migrator implementation
### Description

https://github.com/LiskHQ/lips/pull/381 introduces some minor changes to the following and need to be addressed:
- Update `roundLength` calculation in `ceiling` method
- Calculation for `MIN_CERTIFICATE_HEIGHT`

### Acceptance Criteria

- `ceiling` method is updated as per the latest change
- Config migration takes into consideration the calculation for `MIN_CERTIFICATE_HEIGHT`

### Additional Information

Corresponding issues in Lisk Core:
- https://github.com/LiskHQ/lisk-core/issues/791
- https://github.com/LiskHQ/lisk-core/issues/792

### PR DESCRIPTION
**Pull request by nagdahimanshu**
### What was the problem?

This PR resolves #117 

### How was it solved?

- [x] ceiling method is updated as per the latest change
- [x] Config migration takes into consideration the calculation for `MIN_CERTIFICATE_HEIGHT`
- [x] Update config files for `testnet` and `mainnet`
- [x] Add logic to ask user to stop Lisk Core v3 before starting Lisk Core v4 

### How was it tested?

Local against `testnet`

### CODE REVIEW

**Review by sergeyshemyakov — CHANGES_REQUESTED** [review]
Generally looks good, but there is an error in pos.ts. According to the [LIP](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0063.md#assets-entry-for-pos-module)
```
const r = ceiling(snapshotHeight, ROUND_LENGTH);
```
Should be replaced with:
```
const r = ceiling(snapshotHeight - HEIGHT_PREVIOUS_SNAPSHOT_BLOCK, ROUND_LENGTH);
```

File: src/index.ts [hunk 1]
```diff
@@ -314,16 +318,23 @@ class LiskMigrator extends Command {
...
+						if (isUserConfirmed) {
+							cli.action.start('Starting lisk-core v4');
+							const network = networkConstant.name as string;
+							await startLiskCore(this, finalConfigCorev4, appVersion, liskCorePath, network);
+							this.log('Started Lisk Core v4 at default data directory.');
+							cli.action.stop();
+						}
 					} else {
 						this.log('Skipping auto start Lisk Core Process.');
 					}
```
Comment (sameersubudhi) [inline src/index.ts:340]: The logging should be different for `isLiskCoreV3Stopped` and `isUserConfirmed`, citing the exact reason why we skip ahead.

File: src/utils/config.ts [hunk 2]
```diff
@@ -104,31 +111,32 @@ export const createBackup = async (config: ApplicationConfigV3): Promise<void> =
 export const migrateUserConfig = async (
 	configV3: ApplicationConfigV3,
 	configV4: ApplicationConfig,
+	snapshotHeight: number,
 ): Promise<ApplicationConfig> => {
 	// Assign default version if not available
 	if (!configV4.system.version) {
-		cli.action.start(`Migrating config property 'system' version to: ${DEFAULT_VERSION}.`);
+		cli.action.start(`Migrating config property 'system.version' to: ${DEFAULT_VERSION}.`);
```
Comment (sameersubudhi) [inline src/utils/config.ts:118]: As discussed offline, please update the logs.

**Review by sameersubudhi — CHANGES_REQUESTED** [review]

**Comment by sameersubudhi** [comment]
@sergeyshemyakov Regarding your comment above. As per the implementation in SDK v5.2.2 [here](https://github.com/LiskHQ/lisk-sdk/blob/v5.2.2/framework/src/modules/dpos/rounds.ts#L22-L24), the round number calculation doesn't take into consideration the `genesisHeight` unlike in SDK v6 ([here](https://github.com/LiskHQ/lisk-sdk/blob/v6.0.0-beta.4/framework/src/modules/pos/method.ts#L141-L149)). So, within the migrator, should we not stick to the logic as implemented in SDK v5?


@nagdahimanshu can confirm that updating the invocation from

```
ceiling(a.header.height, roundLength)
```
to
```
ceiling(a.header.height - HEIGHT_PREVIOUS_SNAPSHOT_BLOCK, roundLength)
```
results in error as the snapshot doesn't contain the delegate information for the resulting round number.

**Review by sameersubudhi — APPROVED** [review]
LGTM 👍

**Pull request merged**